### DISCUSSION TITLE
Prisma migrate does not work with Vercel Postgres

### DISCUSSION THREAD

**Discussion by vimtor** [vercel/community#2515] [Bug Report]
Has anyone had success using the Vercel Postgres default configuration?

For me, it works well, but when I use `prisma migrate dev`, I experience problems. With a new database, I can only create the initial migration. After that, I get the following error:

```
The database schema is not empty. Read more about how to baseline an existing production database
```

I've tried different things and creating a separate database for the `shadowDatabaseUrl` fields seems to work.

**Comment by vimtor** [comment]
After some research I found [this page in Neon's docs](https://neon.tech/docs/guides/prisma-migrate) explaining that you need to create a separate database for `prisma migrate` to work.

**Reply by littlegromexchange** [reply]
Did creating a separate database work for you? The vercel integration guide shows to have your `schema.prisa` file like so:
```
datasource db {
  provider          = "postgresql"
  url               = env("POSTGRES_PRISMA_URL")
  shadowDatabaseUrl = env("POSTGRES_URL_NON_POOLING")
  directUrl         = env("POSTGRES_URL_NON_POOLING")
}
```
And vercel only allows 1 db for hobby accounts. Do we need to upgrade to be able to create a separate DB just for the shadow db? The .env var examples at the bottom of the page show two completely different dbs.
```
DATABASE_URL="postgres://daniel:<password>@ep-restless-rice-862380-pooler.us-east-2.aws.neon.tech:5432/neondb?pgbouncer=true"
SHADOW_DATABASE_URL="postgres://daniel:<password>@ep-restless-rice-862380.us-east-2.aws.neon.tech:5432/shadow_db"
DIRECT_URL="postgres://daniel:<password>@ep-restless-rice-862380.us-east-2.aws.neon.tech:5432/neondb"
```

**Reply by vimtor** [reply]
Yes, creating a new DB for the shadow fixed the issue. I guess you can spin up a new DB in you local environment using Docker if you don't want to upgrade just for it.

**Reply by janpio** [reply]
If someone is still reading this, the `shadowDatabaseUrl` is not needed any more: https://github.com/orgs/vercel/discussions/2515#discussioncomment-6450133

**Comment by Radicitus** [comment]
Also having this problem as well! Have my schema.prisma set up exactly like the docs and all the enccessary env vars from vercel. Any solution found? I have a hobby account so I can't make another db.

**Reply by vimtor** [reply]
https://github.com/orgs/vercel/discussions/2515#discussioncomment-5954253

**Comment by douglasrcjames** [comment]
Seems to be a bug with the Beta setup, or at least a lack of forethought into a standard Prisma postgres setup by the Vercel team. Would be nice if the standard spin up of a storage bucket on vercel would allow for a normal database URL and a shadow database URL. 

I upgraded to a Pro plan then reset my credentials hoping it would then provide me with a shadow database URL under the same storage bucket, but it didn't. Don't want to have a messy account with a separate full set of credentials just for the shadow database URL. 

Based on [this thread](https://github.com/prisma/prisma/issues/19234), we can reset the database, but this sounds like it defeats the purpose of the shadow implementation and non-ideal. 

Has anyone found a better work around than this? If not, do we have an open bug report to see if we can get this reconciled by vercel?

**Reply by Radicitus** [reply]
Yes, I believe there is an issue here: https://github.com/vercel/storage/issues/114#issuecomment-1567366440

**Comment by ruheni** [comment]
👋

Vercel Postgres/ Neon recently announced that the need to create the shadow database manually was removed: [Improving the developer experience for Prisma users](https://neon.tech/blog/prisma-dx-improvements). I would recommend removing the `shadowDatabaseUrl` property from your schema and retrying running `prisma migrate dev`.

```diff
datasource db {
  provider  = "postgresql"
  url       = env("DATABASE_URL")
  directUrl = env("DIRECT_DATABASE_URL")
-  shadowDatabaseUrl = env("SHADOW_DATABASE_URL")
}
```

Let us know if it works once you make the update. 🙂

**Comment by S3Prototype** [comment]
Just a general update on Vercel's end: We've rectified our documentation and scrubbed all mentions of `shadowDatabaseUrl`. As @ruheni mentioned, it's not necessary to get set up. Thanks for starting this conversation, folks! 👍

**Reply by mjsilva** [reply]
@S3Prototype `shadowDatabaseUrl` is still on the configuration params for prisma see bellow
![Image 2023-08-13 at 8 56 13 pm](https://github.com/vercel/community/assets/158554/f048a24b-a8e5-400e-860a-7ccf5ed171b0)

Is this not required?

**Reply by janpio** [reply]
The `shadowDatabaseUrl` should not be listed there any more, and is definitely not required. Just delete that line and everything will work.

**Reply by leerob** [reply]
The shadow database URL has been removed 👍
